packages/dds/map/src/test/mocha/map.order.spec.ts

```diff
@@ -0,0 +1,280 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+	MockContainerRuntimeFactory,
+	MockContainerRuntimeFactoryForReconnection,
+	MockContainerRuntimeForReconnection,
+	MockFluidDataStoreRuntime,
+	MockSharedObjectServices,
+	MockStorage,
+} from "@fluidframework/test-runtime-utils";
+import { MapFactory, SharedMap } from "../../map";
+import {
+	IMapClearOperation,
+	IMapDeleteOperation,
+	IMapSetOperation,
+} from "../../internalInterfaces";
+import { TestSharedMap, createConnectedMap, createLocalMap } from "./mapUtils";
+
+describe("Map Iteration Order", () => {
+	describe("Local state", () => {
+		let map: SharedMap;
+
+		beforeEach(async () => {
+			map = createLocalMap("testMap");
+		});
+
+		it("set", () => {
+			map.set("2", 1);
+			map.set("1", 2);
+			map.set("3", 3);
+
+			assert.deepEqual(Array.from(map.keys()), ["2", "1", "3"]);
+			assert.deepEqual(Array.from(map.values()), [1, 2, 3]);
+		});
+
+		it("set with value overwritting", () => {
+			map.set("2", 1);
+			map.set("1", 2);
+			map.set("3", 3);
+			map.set("1", 4);
+			map.set("4", 5);
+			map.set("2", 6);
+
+			assert.deepEqual(Array.from(map.keys()), ["2", "1", "3", "4"]);
+			assert.deepEqual(Array.from(map.values()), [6, 4, 3, 5]);
+		});
+
+		it("delete", () => {
+			map.delete("3");
+			map.set("2", 1);
+			map.set("1", 2);
+			map.set("3", 3);
+
+			map.delete("1");
+			assert.deepEqual(Array.from(map.keys()), ["2", "3"]);
+
+			map.set("1", 4);
+			assert.deepEqual(Array.from(map.keys()), ["2", "3", "1"]);
+			assert.deepEqual(Array.from(map.values()), [1, 3, 4]);
+		});
+
+		it("clear", () => {
+			map.set("2", 1);
+			map.set("1", 2);
+			map.set("3", 3);
+
+			map.clear();
+			assert.deepEqual(Array.from(map.keys()), []);
+			assert.deepEqual(Array.from(map.values()), []);
+
+			map.set("3", 1);
+			map.set("2", 2);
+			map.delete("3");
+			map.set("1", 3);
+
+			assert.deepEqual(Array.from(map.keys()), ["2", "1"]);
+		});
+
+		it.skip("serialize/load", async () => {
```

Comment:
glad to see this added. Once you settle some more problems with serialization, it'd be good to dedicate a few cases to it. I'd want some coverage for the following aspects:

- Verify compat issues with the existing format (what happens when we load existing summary?)
- Verify summaries with/without multiple blobs work well